semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/TypeOps.scala

```diff
@@ -56,36 +53,8 @@ trait TypeOps { self: Scalacp =>
                 s.TypeRef(s.NoType, ssym, sargs)
             }
           case ConstantType(const) =>
-            def floatBits(x: Float) = java.lang.Float.floatToRawIntBits(x).toLong
-            def doubleBits(x: Double) = java.lang.Double.doubleToRawLongBits(x)
-            const match {
-              case () =>
-                s.SingletonType(st.UNIT, s.NoType, Symbols.None, 0, "")
-              case false =>
-                s.SingletonType(st.BOOLEAN, s.NoType, Symbols.None, 0, "")
-              case true =>
-                s.SingletonType(st.BOOLEAN, s.NoType, Symbols.None, 1, "")
-              case x: Byte =>
-                s.SingletonType(st.BYTE, s.NoType, Symbols.None, x.toLong, "")
-              case x: Short =>
-                s.SingletonType(st.SHORT, s.NoType, Symbols.None, x.toLong, "")
-              case x: Char =>
-                s.SingletonType(st.CHAR, s.NoType, Symbols.None, x.toLong, "")
-              case x: Int =>
-                s.SingletonType(st.INT, s.NoType, Symbols.None, x.toLong, "")
-              case x: Long =>
-                s.SingletonType(st.LONG, s.NoType, Symbols.None, x, "")
-              case x: Float =>
-                s.SingletonType(st.FLOAT, s.NoType, Symbols.None, floatBits(x), "")
-              case x: Double =>
-                s.SingletonType(st.DOUBLE, s.NoType, Symbols.None, doubleBits(x), "")
-              case x: String =>
-                s.SingletonType(st.STRING, s.NoType, Symbols.None, 0, x)
-              case null =>
-                s.SingletonType(st.NULL, s.NoType, Symbols.None, 0, "")
-              case other =>
-                sys.error(s"unsupported const $other")
-            }
+            val sconst = s.Constant(const)
+            s.ConstantType(sconst)
```

Comment:
😍 Thanks for the idea to split constant types even further!

Reply:
I'm glad it came out nicely 👍